### PR TITLE
Update knative-serving docs to match new template #4614

### LINKED ISSUE #4614: Update knative-serving package documentation
## Feature Request

There is an effort to update and standardize package documentation. Update the current knative-serving package (1.0.0) with this updated template.

### PR DESCRIPTION
**Pull request by seemiller**
## What this PR does / why we need it

This pull request updates the knative-serving documentation to follow an updated template for package documentation.

## Which issue(s) this PR fixes

Fixes: #4614

### CODE REVIEW

File: addons/packages/knative-serving/1.0.0/README.md [hunk 1]
```diff
@@ -1,91 +1,115 @@
...
+
+```shell
+cat <<EOF >knative-values.yaml
+domain:
+  type: real
+  name: example.com
+EOF
+```
+
+Install Knative-Serving
```
Comment (LappleApple) [inline addons/packages/knative-serving/1.0.0/README.md:27]: Add period here to be consistent ...

File: addons/packages/knative-serving/1.0.0/README.md [hunk 2]
```diff
@@ -1,91 +1,115 @@
...
-The Knative Serving package requires use of Contour for ingress. To successfully install and use the Knative Serving package, you must first install Contour.
+There are no additional application configuration options that can be passed to the Knative-Serving application at this time.
 
-```shell
-tanzu package install contour --package-name contour.community.tanzu.vmware.com --version 1.19.1
-```
+## What This Package Does
 
-After the Contour package has been installed, you can install Knative-Serving.
+Knative Serving provides components that enable rapid deployment of serverless containers and autoscaling of pods.
```
Comment (LappleApple) [inline addons/packages/knative-serving/1.0.0/README.md:55]: Add hyphen in "Knative Serving" to be consistent with references elsewhere in the doc

File: addons/packages/knative-serving/1.0.0/README.md [hunk 3]
```diff
@@ -95,15 +119,9 @@ a096a354b2818471e91c583ff2fb01f5-975360063.us-east-1.elb.amazonaws.com
     EOF
     ```
 
-1. Install the knative-serving package.
-
-    ```shell
-    tanzu package install knative-serving --package-name knative-serving.community.tanzu.vmware.com --version 1.0.0
-    ```
+2. Install the knative-serving package. Refer to the [installation](#installation) section for more details.
```
Comment (LappleApple) [inline addons/packages/knative-serving/1.0.0/README.md:122]: Capitalize knative-serving

File: addons/packages/knative-serving/1.0.0/README.md [hunk 4]
```diff
@@ -129,23 +147,23 @@ a096a354b2818471e91c583ff2fb01f5-975360063.us-east-1.elb.amazonaws.com
...
 
-1. In a separate terminal window, watch the pods.
+5. In a separate terminal window, watch the pods.
 
     ```shell
     watch kubectl get pods --namespace example
     ```
 
-1. At this point you will have a Magic DNS (nip.io) configured to provide you access to your service. If you want to use real DNS or any other Magic DNS you'll need to provide the appropriate configuration to the knative-serving package when installing it.
+6. At this point you will have a [Magic DNS](https://nip.io) configured to provide you with access to your service. If you want to use real DNS or any other Magic DNS you'll need to provide the appropriate configuration to the knative-serving package when installing it.
```
Comment (LappleApple) [inline addons/packages/knative-serving/1.0.0/README.md:162]: Capitalize "knative-serving"

**Review by lasomethingsomething — COMMENTED** [review]
Thanks @seemiller, great work!

**Pull request merged**